### PR TITLE
dockerization, code cleanup

### PR DESCRIPTION
**Pull request by yaleman**
I can't actually test the Slack bit because I don't have a token, but it starts up and connects to MISP OK...

# New files

- `Dockerfile` - used to build the image
- `docker-compose.yml` - Sets up volumes etc
- `.pylintrc` - skips C0301 which is "line too long" in pylint

# Edits

`helper.py` 
- general cleanup to make pylint happy

`mispattruploader.py` 
- cleanup, had a few vague variable names and if/else syntax curiosities

`main.py` 
- reformatted to set up logging earlier and cleaner
- handles missing config options better and sets logging defaults

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+# Make a new directory to put our code in.
+RUN mkdir /code
+
+WORKDIR /code
+
+COPY requirements.txt /code/
+RUN pip install -r requirements.txt
+
+COPY *.py /code/
+COPY config.json /code/
```
Comment (mchaffe) [inline Dockerfile:17]: Baking in the config is not secure. Leave it out and include it via volume

File: helper.py [hunk 2]
```diff
@@ -1,18 +1,22 @@
...
+            user = user_info['user']
+            print('ok')
+            if user.get('profile') is not None:
+                profile = user['profile']
+                if profile.get('display_name') is not None:
+                    print('ok')
+                    username = profile['display_name']
+                    print('Returning %s' %username)
+                    return username
+    return False
```
Comment (mchaffe) [inline helper.py:22]: Should be using logger instead of pprint

File: main.py [hunk 3]
```diff
@@ -1,154 +1,210 @@
...
+                       misp_key=data['misp']['key'],
+                       misp_ssl=data.get('misp', {}).get('ssl', True), # default to using SSL
+                       )
+    logger.info("Connected to misp server successfully")
+# Who knows what kind of errors PyMISP will throw?
+#pylint: disable=broad-except
+except Exception:
+    logger.error('Failed to connect to MISP:')
+    logger.error(traceback.format_exc())
+    sys.exit()
```
Comment (mchaffe) [inline main.py:109]:
```suggestion
    sys.exit(1)
```

File: main.py [hunk 4]
```diff
@@ -1,154 +1,210 @@
...
+
+# config file - slack section
+if not data.get('slack'):
+    logger.error("No 'slack' config section, quitting.")
+    sys.exit()
```
Comment (mchaffe) [inline main.py:114]:
```suggestion
    sys.exit(1)
```

File: main.py [hunk 5]
```diff
@@ -1,154 +1,210 @@
...
+
+MISSED_SLACK_KEY = False
+for key in ('SLACK_BOT_OAUTH_TOKEN', 'SLACK_SIGNING_SECRET'):
+    if key not in data.get('slack'):
+        MISSED_SLACK_KEY = True
+        logger.error("Couldn't find %s in config.json slack section, going to quit.", key)
+if MISSED_SLACK_KEY:
+    sys.exit()
```
Comment (mchaffe) [inline main.py:122]:
```suggestion
    sys.exit(1)
```

File: main.py [hunk 6]
```diff
@@ -1,154 +1,210 @@
...
+    # shouldn't get here, but return a 403 if you do.
+    return_value = 'Unhandled message type', 403
+    return return_value
+
 @slack_events_adapter.on("error")
 def error_handler(err):
-	logger.info('got error from slack')
-	print("ERROR: " + str(err))
+    """ slack error message handler """
+    logger.error("Slack error: %s", str(err))
```
Comment (mchaffe) [inline main.py:190]:
```suggestion
    logger.error(f"Slack error: {err}")
```

File: Dockerfile (hunk 1 above)
Comment (yaleman) [reply Dockerfile:17]: I'd totally agree, would prefer to do it in environment variables but didn't want to mess with workflows being new to this project 👍🏻

File: main.py (hunk 3 above)
Comment (yaleman) [reply main.py]: Fixed in a3c3849

File: main.py (hunk 5 above)
Comment (yaleman) [reply main.py]: Fixed in 4bb571c

File: main.py (hunk 6 above)
Comment (yaleman) [reply main.py:190]: Best practice is to use the lazy evaluation as it is.

File: main.py (hunk 4 above)
Comment (yaleman) [reply main.py]: Fixed in 4bb571c

File: helper.py (hunk 2 above)
Comment (yaleman) [reply helper.py]: done in 50b9af7

**Comment by parker67** [comment]
Looks good to go.

**Pull request merged**